test/lib/storage_proposal_test.rb

```diff
@@ -6,84 +6,57 @@
-      double("storage_manager", y2storage_probed: devicegraph, probed_disk_analyzer: disk_analyzer)
-    end
-
-    before do
-      allow(Y2Storage::StorageManager).to receive(:instance)
-        .and_return(storage_manager)
-      allow(Y2Storage::ProposalSettings).to receive(:new_for_current_product)
-        .and_return(proposal_settings)
-      allow(Y2Storage::GuidedProposal).to receive(:new)
-        .and_return(guided_proposal)
-      allow(Y2Storage::AutoinstProposal).to receive(:new)
-        .and_return(autoinst_proposal)
-      allow(storage_manager).to receive(:proposal=)
-      allow(Y2Storage::AutoinstProposal).to receive(:new).and_return(autoinst_proposal)
-      allow(Y2Storage::DiskAnalyzer).to receive(:new).with(devicegraph).and_return(disk_analyzer)
-    end
+  let(:storage_manager) { double(Y2Storage::StorageManager) }
+  let(:guided_proposal) { instance_double(Y2Storage::GuidedProposal, propose: nil, proposed?: true) }
+  let(:autoinst_proposal) { instance_double(Y2Storage::AutoinstProposal, propose: nil, proposed?: true) }
+
+  before do
+    allow(Y2Storage::StorageManager).to receive(:instance)
+      .and_return(storage_manager)
+    allow(Y2Storage::GuidedProposal).to receive(:initial)
+      .and_return(guided_proposal)
+    allow(Y2Storage::AutoinstProposal).to receive(:new)
+      .and_return(autoinst_proposal)
+    allow(storage_manager).to receive(:proposal=)
+  end
 
+  describe "#initialize" do
     context "when no partitioning plan is given" do
       let(:profile) { nil }
 
-      it "sets a guided proposal" do
-        expect(Y2Storage::GuidedProposal).to receive(:new)
-          .with(settings: proposal_settings, devicegraph: devicegraph, disk_analyzer: disk_analyzer)
-        expect(storage_manager).to receive(:proposal=).with(guided_proposal)
-        storage_proposal.propose_and_store
-      end
-
-      it "returns true" do
-        expect(subject.propose_and_store).to eq(true)
-      end
-
-      context "if proposal fails" do
-        before do
-          allow(guided_proposal).to receive(:propose).and_raise(Y2Storage::Error)
-        end
-
-        it "does not set the proposal" do
-          expect(storage_manager).to_not receive(:proposal=)
-          storage_proposal.propose_and_store
-        end
-
-        it "returns false" do
-          expect(storage_manager).to_not receive(:proposal=)
-          storage_proposal.propose_and_store
-        end
+      it "creates a guided proposal" do
+        expect(Y2Storage::GuidedProposal).to receive(:initial)
+        expect(storage_proposal.proposal).to be(guided_proposal)
       end
     end
 
     context "when profile contains an empty set of partitions" do
       let(:profile) { [] }
 
-      it "sets a proposal" do
-        expect(storage_manager).to receive(:proposal=).with(guided_proposal)
-        storage_proposal.propose_and_store
+      it "creates a guided proposal" do
+        expect(Y2Storage::GuidedProposal).to receive(:initial)
```

Comment:
I think that is because these lines in the `before` block:

```
allow(Y2Storage::GuidedProposal).to receive(:initial)
      .and_return(guided_proposal)
```